central/detection/alertmanager/alert_manager_impl.go

```diff
@@ -166,10 +166,12 @@ func (d *alertManagerImpl) notifyAndUpdateBatch(ctx context.Context, alertsToMar
 	return d.updateBatch(ctx, alertsToMark)
 }
 
-// It is the caller's responsibility to not call this with an empty slice,
-// else this function will panic.
-func lastTimestamp(processes []*storage.ProcessIndicator) *ptypes.Timestamp {
-	return processes[len(processes)-1].GetSignal().GetTime()
+// It is the caller's responsibility to not call this with an empty slice.
+func lastTimestamp(processes []*storage.ProcessIndicator) (*ptypes.Timestamp, error) {
+	if len(processes) == 0 {
```

Comment:
I'm not knowledgable of Go, but is it still possible for a calling function to pass a non-array at runtime, that would still cause `len` to throw, or would that be caught at compile time?